src/components/App.tsx

```diff
@@ -1,14 +1,113 @@
+    const [searchedInput, setSearchedInput] = useState("");
+    const [fetchedEpisodes, setFetchedEpisodes] = useState<IEpisode[]>([]);
+    const [loading, setLoading] = useState(true);
+    const [currentShow, setCurrentShow] = useState(showsData[0]);
+
+    const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setSearchedInput(event.target.value);
+    };
+
+    const handleSelectShow = (e: React.ChangeEvent<HTMLSelectElement>) => {
+        setCurrentShow(showsData.find((show) => show.name === e.target.value)!);
+        setSearchedInput("");
+    };
+
+    useEffect(() => {
+        async function getDataFromAPI() {
+            const response = await fetch(
+                "https://api.tvmaze.com/shows/" + currentShow.id + "/episodes"
+            );
+            const jsonBody: IEpisode[] = await response.json();
+            jsonBody.forEach((episode: IEpisode) => {
+                if (!episode.image) {
+                    episode.image = {
+                        medium: "",
+                        original: "",
+                    };
+                }
+                if (!episode.summary) {
+                    episode.summary = "";
+                }
+            });
+            setFetchedEpisodes(jsonBody);
+            setLoading(false);
+        }
+
+        //reinsuring that this is run only once, on mount.
+        getDataFromAPI();
+    }, [currentShow]);
+
+    const allEpisodes = filterBySearchedInput(
+        fetchedEpisodes,
+        searchedInput
+    ).map((episode) => <EpisodeCard key={episode.id} episode={episode} />);
+
     return (
-        <div className="App">
-            <MyComponent />
+        <>
+            <div className="title">TV Shows</div>
+            <div className="searchBar">
+                <span className="series-selector">
+                    <label htmlFor="series">Choose a Show: </label>
 
-            {greet("World")}
-        </div>
+                    <select
+                        name="series"
+                        id="series"
+                        onChange={handleSelectShow}
+                    >
+                        {showsData.map((show) => (
+                            <option key={show.id} value={show.name}>
+                                {show.name}
+                            </option>
+                        ))}
+                    </select>
+                </span>
+                <span className="search-annotation">Search episodes:</span>
+                <KeyboardControlledInput
+                    value={searchedInput}
+                    onChange={handleInputChange}
+                />
+                <span className="counter">
+                    Showing:
+                    {
+                        filterBySearchedInput(fetchedEpisodes, searchedInput)
+                            .length
+                    }
+                    /{fetchedEpisodes.length}
+                </span>
+            </div>
```

Comment:
This could be a seperate component - and contain KeyboardControlledInput (or have it seperate)